models/articles.model.js

```diff
@@ -1,5 +1,43 @@
 const db = require("../db/connection");
 
+exports.insertComment = (article_id, username, body) => {
+  if (!username || !body) {
+    return Promise.reject({ status: 400, msg: "400: Bad Request - Missing required fields" });
+  }
+
+  return db
+    .query(
+      `SELECT 1 FROM articles WHERE article_id = $1`,
+      [article_id]
+    )
+    .then((result) => {
+      if (result.rowCount === 0) {
+        return Promise.reject({ status: 404, msg: "404: Article Not Found" });
+      }
+      return db.query(
+        `SELECT 1 FROM users WHERE username = $1`,
+        [username]
+      );
+    })
+    .then((userResult) => {
+      if (userResult.rowCount === 0) {
+        return Promise.reject({ status: 404, msg: "404: User Not Found" });
+      }
+      
```

Comment:
This could be DRYer, maybe move some of these functions to your controller, are some of these db.queries already being used in other functions that you could re-use?